salt/metalk8s/container-engine/containerd/installed.sls

```diff
@@ -36,12 +37,17 @@ Install containerd:
 Create containerd service drop-in:
   file.managed:
     - name: /etc/systemd/system/containerd.service.d/50-metalk8s.conf
-    - source: salt://{{ slspath }}/files/50-metalk8s.conf
+    - source: salt://{{ slspath }}/files/50-metalk8s.conf.j2
+    - template: jinja
     - user: root
     - group: root
     - mode: 0644
     - makedirs: true
     - dir_mode: 0755
+    - context:
+      environment: >-
+        NO_PROXY=127.0.0.1,localhost,.svc,.default,.local,{{
```

Comment:
I don't know the allowed syntax of `NO_PROXY`, but this will probably won't be enough.

First question: what's the exact symptom? When we have a `HTTP_PROXY` set up, is `containerd` failing on retrieval of new container images? Are containers also failing to reach other services/cluster IPs?

Second question: if `containerd` uses the proxy for container networking as well, then we'll need a lot more than this (I don't even think we could be exhaustive) - all Service DNS names can use `.svc.<namespace>`, or `.svc.<namespace>.cluster`, or `.svc.<namespace>.cluster.local`, or even remove this whole suffix and only use the Service actual name. We naturally cannot specify all of them here, so maybe we could instead try to resolve names with CoreDNS before trying to go through the proxy?

Reply:
During my tests, only containerd was failing to retrieve container images.
I don't think that these environment variables are used for running containers (at least for cluster internal communication)

Reply:
OK, then I guess we don't need all those rules, just the registry address, no? Something like `metalk8s-registry-from-config.invalid` (or its actual IP:port, not sure)?

Reply:
I seem to remember from the Slack chats with TS, the processes *inside* containers were also using the proxy variables, then failing to connect to e.g. `10.96.0.1` or whatnot. Please validate this more in-depth...

In any case:
- Where does `.default` come from?
- Where does `.local` come from?
- Indeed, what about the registry 'DNS name'?

Reply:
I think that only the registry address could be sufficient, when I try to deploy setting only this address in the no_proxy variable, it seems to work fine.
Maybe adding the whole network would be a better idea, just in case we want to handle multiple replicas for registry.
As I said above, processes inside containers don't seem to be affected by these environment variables.